File: LibraryManagementSystem/Frontend/library-management-system-web/src/app/entities/dtos/library/filter-monograph-dto.ts
import { CareerDto } from "../university/career-dto";
import { AuthorDto } from "./author-dto";

export interface FilterMonographDto {
    authors: AuthorDto[] | null;
    careers: CareerDto[] | null;
    beginPresentationDate: Date | null;
    endPresentationDate: Date | null;
}

File: LibraryManagementSystem/Backend/LibraryManagementSystem/LibraryManagementSystem.Entities/Dtos/Library/FilterMonographDto.cs
using LibraryManagementSystem.Entities.Dtos.University;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LibraryManagementSystem.Entities.Dtos.Library
{
    public class FilterMonographDto
    {
        // Authors
        [Required]
        [JsonPropertyName("authors")]
        public IEnumerable<AuthorDto>? Authors { get; set; }

        // Authors
        [Required]
        [JsonPropertyName("careers")]
        public IEnumerable<CareerDto>? Careers { get; set; }

        [Required]
        [JsonPropertyName("beginPresentationDate")]
        public DateTime? BeginPresentationDate { get; set; }

        [Required]
        [JsonPropertyName("endPresentationDate")]
        public DateTime? EndPresentationDate { get; set; }
    }
}
